File: csharp/grade-school/GradeSchool.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class GradeSchool
{
    private static Dictionary<string, int> students = new Dictionary<string, int>();
    public void Add(string student, int grade)
    {
        students.Add(student, grade);
    }

    public IEnumerable<string> Roster()
    {
        var result = students
                    .OrderBy(s => s.Value)
                    .ThenBy(s => s.Key)
                    .Select(s => s.Key)
                    .ToArray();
        students.Clear();
        return result;
    }

    public IEnumerable<string> Grade(int grade)
    {
        var result = students
                    .Where(s=>s.Value == grade)
                    .OrderBy(s => s.Key)
                    .Select(s => s.Key)
                    .ToArray();
        students.Clear();
        return result;
    }
}

File: javascript/grade-school/grade-school.js
export class GradeSchool {
  constructor() {
    this._roster = {};

  }
  roster = () => JSON.parse(JSON.stringify(this._roster));
  grade = (grade) => [...(this._roster[grade] || [])];

  add = (student, grade) => {
    this._roster[grade] = this._roster[grade] || [];
    this._roster[grade].push(student);
    this._roster[grade].sort();
      
    }
  }


